### PR TITLE
Update examples to use wp-scripts 5.0.0

### LINKED ISSUE #88: build php dependencies etc. are not used
Why are these not used in the `wp_register_script` function? https://github.com/WordPress/gutenberg-examples/blob/b74542f4b8e31bab4a34d51a928279e92055f5ce/01-basic-esnext/build/index.asset.php#L1

### LINKED ISSUE #90: Update import to use standard syntax
Switch ESNext examples to use `import { .. } from '@wordpress/package'` syntax
Fix deprecation warnings for editor => blockEditor and others.

### PR DESCRIPTION
**Pull request by mkaz**
Let's do all the updates... 

- Update examples package to use wp-scripts version 5.0.0
- Update examples to use index.assets.php file to auto load the dependencies and version
- Switch syntax to use `import { ... } from '@wordpress/package'`

Additionally, I removed the check for `register_block_type` function, since it is available in WP 5.0+. I don't think it is still necessary to show an example for it.

Fixes #88 
Fixes #90 
Also probably makes PR #87 obsolete.

### CODE REVIEW

**Comment by mkaz** [comment]
Since this was a large set of changes I additional bumped the plugin versions to `1.1.0`
I'm not really sure if it makes much difference.

File: 01-basic-esnext/src/index.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /* global wp */
```
Comment (gziolo) [inline 01-basic-esnext/src/index.js]: This comment for ESLint can be removed here and in other files since we no longer use them

Comment (mkaz) [reply 01-basic-esnext/src/index.js]: Good catch. Removed

**Comment by gziolo** [comment]
🎉 

> Also probably makes PR #87 obsolete.

Yes, I will close that one 👍

**Comment by gziolo** [comment]
I missed one thing when doing review. Those import statements should have the corresponding entries in package.json file in “dependencies” section.

**Comment by gziolo** [comment]
How do we communicate that those import statements should have the corresponding entries in package.json file in “dependencies” section?

We can include ESLint rule which helps with validation of whether imported dependencies are listed in package.json file. Although there is some bug in ESLint which mades its usage in wp-scripts complicated...

**Pull request merged**